### PR TITLE
Vicky/make album api

### PR DESCRIPTION
**Pull request by vickyban**
changed image api, now initial data from ImgurMainGallery will be stored in album object, each contains a list of image objects

### CODE REVIEW

File: src/lib/imgur.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  // page - integer
+
+  // hot+viral = popular
+  //hot+time = newest
+  async  fetchMainGallery(section = "hot", sort = "viral", page = 0) {
+    const uri = `https://api.imgur.com/3/gallery/${section}/${sort}/${page}.json`;
+
+    let header = new Headers();
+    header.append("Authorization", `Client-ID ${CLIENT_ID}`);
+    let request = new Request(uri, { meathod: "GET", headers: header })
```
[the remark this answers is not in the document]
Comment (vickyban) [reply src/lib/imgur.js]: done

File: src/lib/imgur.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+    let jsonObj = await fetch(request).then(res => res.json())
+
+    let formattedAlbums = jsonObj.data.map(data => this.formatAlbum(data))
```
[the remark this answers is not in the document]
Comment (vickyban) [reply src/lib/imgur.js]: that didn't work

File: .gitignore [hunk 3]
```diff
@@ -3,6 +3,9 @@
 # dependencies
 /node_modules
 
+# api keys
+src/*/config_key.js
```
Comment (FaizanMinhas) [inline .gitignore]: `config_key.js` is what this should be

File: src/pages/homePage.js [hunk 4]
```diff
@@ -1,14 +1,46 @@
...
+      let albumDivs = albums.map(album => {
+        return <div className='album_div' key={album.id} >
+          <Link to={`/image/${album.id}`}>
+            <Image id={album.images[0].id} link={album.images[0].link}></Image>
+          </Link>
+        </div>
+      })
+      return albumDivs;
+    }
+    return null;
```
Comment (FaizanMinhas) [inline src/pages/homePage.js]: Actually let's refactor this to only have 1 return. 
declare albumDivs at the top i.e. `let albumDivs = null`
And then return albumDivs at the bottom. This should make the code intent more clear.

**Pull request merged**